File: app/src/main/java/mobiletest/yucef/ca/mobiletest/model/Address.java
package mobiletest.yucef.ca.mobiletest.model;

public class Address {
    String address1;
    String label;
    String zipCode;
    String city;
    String state;
    String country;
    Long latitude;
    Long longitude;

    public Address(String address1, String label, String zipCode, String city, String state, String country, Long latitude, Long longitude) {
        this.address1 = address1;
        this.label = label;
        this.zipCode = zipCode;
        this.city = city;
        this.state = state;
        this.country = country;
        this.latitude = latitude;
        this.longitude = longitude;
    }

    public String getAddress1() {
        return address1;
    }

    public String getLabel() {
        return label;
    }

    public String getZipCode() {
        return zipCode;
    }

    public String getCity() {
        return city;
    }

    public String getState() {
        return state;
    }

    public String getCountry() {
        return country;
    }

    public Long getLatitude() {
        return latitude;
    }

    public Long getLongitude() {
        return longitude;
    }
}

File: app/src/main/java/com/quickseries/mobiletest/domain/resources/model/Address.kt
package com.quickseries.mobiletest.domain.resources.model

import com.quickseries.mobiletest.data.resources.AddressResponse
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext

data class Address(
    val address1: String,
    val label: String,
    val zipCode: String,
    val city: String,
    val state: String,
    val country: String,
    val latitude: String,
    val longitude: String
) {
    /**
     * Simple function to determine fi all the fields of the data class are empty.
     *
     * @return Boolean true if empty
     */
    fun isEmpty() =
        address1.isEmpty() && label.isEmpty() && zipCode.isEmpty() && city.isEmpty() && state.isEmpty() && country.isEmpty() && latitude.isEmpty() && longitude.isEmpty()
}

/**
 * Extension that convert a list of [AddressResponse] to a list of [Address].
 *
 * @return List<Address>
 */
suspend fun List<AddressResponse>.toAddresses() = withContext(Dispatchers.Default) {
    map { addressResponse ->
        Address(
            addressResponse.address1.orEmpty(),
            addressResponse.label.orEmpty(),
            addressResponse.zipCode.orEmpty(),
            addressResponse.city.orEmpty(),
            addressResponse.state.orEmpty(),
            addressResponse.country.orEmpty(),
            addressResponse.gps.latitude.orEmpty(),
            addressResponse.gps.longitude.orEmpty()
        )
    }
}